pystac_client/conformance.py

```diff
@@ -17,93 +17,96 @@
-        return uri == self.uri
-
-
-STAC_PREFIX = 'https://api.stacspec.org/v1.0.0-beta.1'
-STAC_PREFIX_LEGACY = 'http://stacspec.org/spec/api/1.0.0-beta.1'
-
-STAC_API_CORE = ConformanceClass(name='STAC API - Core',
-                                 uri=f'{STAC_PREFIX}/core',
-                                 alternative_uris={f'{STAC_PREFIX_LEGACY}/core'})
-"""Used to test conformance with the `STAC API - Core spec
-<https://github.com/radiantearth/stac-api-spec/tree/master/core>`__."""
-
-STAC_API_ITEM_SEARCH = ConformanceClass(name='STAC API - Item Search',
-                                        uri=f'{STAC_PREFIX}/item-search',
-                                        alternative_uris={f'{STAC_PREFIX_LEGACY}/req/stac-search'})
-"""Used to test conformance with the `STAC API - Item Search spec
-<https://github.com/radiantearth/stac-api-spec/tree/master/item-search>`__."""
-
-STAC_API_ITEM_SEARCH_CONTEXT_EXT = ConformanceClass(
-    name='STAC API - Item Search: Context Extension',
-    uri=f'{STAC_API_ITEM_SEARCH.uri}#context',
-    alternative_uris={f'{STAC_PREFIX_LEGACY}/req/context'})
-"""Used to test conformance with the `Context Extension
-<https://github.com/radiantearth/stac-api-spec/tree/master/fragments/context>`__ to the STAC API - Item Search"""
-
-
-class ConformanceClasses:
-    """Enumerates the conformance classes used by this package."""
-    STAC_API_CORE = STAC_API_CORE
-    STAC_API_ITEM_SEARCH = STAC_API_ITEM_SEARCH
-    STAC_API_ITEM_SEARCH_CONTEXT_EXT = STAC_API_ITEM_SEARCH_CONTEXT_EXT
+    def conformance(self) -> List[str]:
+        """Overwrite in the sub-class to list the conformance URIs for this object."""
+        return self._conformance
+
+    @conformance.setter
+    def conformance(self, value):
+        self._conformance = value
+
+    def conforms_to(self, key: str) -> bool:
+        """Whether the API conforms to the given standard. This method only checks against the ``"conformsTo"``
+        property from the API landing page and does not make any additional calls to a ``/conformance`` endpoint
+        even if the API provides such an endpoint.
+
+        Parameters
+        ----------
+        spec : str or ConformanceClass
+            Either a :class:`~pystac_client.conformance.ConformanceClass` instance or the URI string for the spec.
+
+        Returns
+        -------
+        bool
+            Indicates if the API conforms to the given spec or URI.
+        """
+
+        # Conformance of None means ignore all conformance as opposed to an
+        #  empty array which would indicate the API conforms to nothing
+        if self.conformance is None:
+            return True
+
+        conformance_class = CONFORMANCE_CLASSES[key]
+
+        def check_conformance():
+            for uri in conformance_class["uris"]:
+                if uri in self.conformance:
+                    return True
+            return False
+
+        if not check_conformance():
```

Comment:
Could simplify this to:

```suggestion
        if not any(uri in conformance_class["uris"] for uri in self.conformance):
```

if you want.

Reply:
:+1: